### PR TITLE
🎉 Source Mailchimp: support oauth flow

### LINKED ISSUE #6273: Source Mailchimp: support oauth
## Tell us about the problem you're trying to solve
With the release of Airbyte Cloud, we need to start supporting Oauth for this connector, since it's the recommended way of authenticating users into a SaaS application. 

If this connector doesn't support oauth already (i.e: doesn't accept a client_id and client_secret) then we need to update its spec to accept those parameters. There are two ways to do this: 

**If the connector already supports some auth mechanism like api_key**, I suggest that this be a oneof nested inside a top-level field called "authentication": 

`
{
  authentication: {
     type: object
     oneOf: [ 
       // api key,
       // oauth
     ] 
  }
}
`

**If the connector only supports webflow oauth**, then no changes are needed to the properties format and we will only need to add annotations. 

See the connector spec reference in the docs for reference on how a oneof can be implemented.

This should be done in a backwards compatible manner i.e: users currently supplying authentication info in the config's top-level should not be impacted by this change. 

## Acceptance Criteria

1. The connector supports oauth webflow authentication with client_id/client_secret in a backwards compatible manner
2. Oauth properties are annotated properly. See [this PR](https://github.com/airbytehq/airbyte/pull/6022) for an example

- [x] #7406

### PR DESCRIPTION
**Pull request by annalvova05**
## What
Add Mailchimp OAuth support.
mailchimp_1
mailchimp_2
mailchimp_3

Java oAuth tests are passed.
mailchimp_4

## How
Change spec.json and source.py

## Recommended reading order
1. `spec.json`
2. `source.py`
3. `MailchimpOAuthFlow.java`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/airbytehq/airbyte/7159)

### CODE REVIEW

File: airbyte-integrations/connectors/source-mailchimp/integration_tests/invalid_config.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 {
-  "apikey": "api-key-awesome",
-  "username": "fakemaster@silly.com"
+  "credentials": {
```
Comment (midavadim) [inline airbyte-integrations/connectors/source-mailchimp/integration_tests/invalid_config.json:2]: Please leave this config without changes and add new ouath_invalid_config.json

We need to check if connector still can work with old config

Comment (annalvova05) [reply airbyte-integrations/connectors/source-mailchimp/integration_tests/invalid_config.json]: Leave old `invalid_config.json` and add with new credentials format `invalid_config_apikey.json.`

**Review by midavadim — APPROVED** [review]

**Comment by antixar** [comment]
@annalvova05 , Please publish your changes

**Pull request merged**